src/main/java/org/zalando/nakadi/util/SubscriptionsUriHelper.java

```diff
@@ -0,0 +1,50 @@
+package org.zalando.nakadi.util;
+
+import org.zalando.nakadi.domain.PaginationLinks;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static java.text.MessageFormat.format;
+
+public class SubscriptionsUriHelper {
+
+    public static PaginationLinks createSubscriptionPaginationLinks(final Optional<String> owningApplication,
+                                                                    final Set<String> eventTypes, final int offset,
+                                                                    final int limit, final int currentPageItemCount) {
+        Optional<PaginationLinks.Link> prevLink = Optional.empty();
+        if (offset > 0) {
+            final String prevUri = createSubscriptionListUri(owningApplication, eventTypes, offset - 1, limit);
+            prevLink = Optional.of(new PaginationLinks.Link(prevUri));
+        }
+
+        Optional<PaginationLinks.Link> nextLink = Optional.empty();
+        if (currentPageItemCount >= limit) {
+            final String nextUri = createSubscriptionListUri(owningApplication, eventTypes, offset + 1, limit);
+            nextLink = Optional.of(new PaginationLinks.Link(nextUri));
+        }
+
+        return new PaginationLinks(prevLink, nextLink);
+    }
+
+    public static String createSubscriptionListUri(final Optional<String> owningApplication,
+                                                   final Set<String> eventTypes, final int offset, final int limit) {
+        final StringBuilder urlBuilder = new StringBuilder("/subscriptions?");
+        if (!eventTypes.isEmpty()) {
```

Comment:
I think you do not need this check here because if Stream is empty it wont do opearations which are defined on it